app/models/user.rb

```diff
@@ -48,6 +48,12 @@ class User < ApplicationRecord
   has_many :user_mail_notifications, :dependent => :destroy, :inverse_of => :user
   has_many :mail_notifications, :through => :user_mail_notifications
   has_many :notification_recipients, :dependent => :delete_all
+  has_many :external_usergroups, lambda {
+    ugroup_at = UsergroupMember.arel_table
```

Comment:
This should most likely be using cached usergroup membership, user groups can be nested, e.g. if UG a contains UG b, all users from b are also in a. This is cached via CachedUsergroupMember model. Also I'm not sure why we can't join user's authsource and use it in SQL condition. Do we need to use Arel here? I have a feeling AR might be enough.

Reply:
Thanks for head up.
But no, this should actually not be using the CachedUsergroup as external usergroup is purely the mapping model.
It's scope of ExternalUsergroup, joining all the way to user is imposible because of the polymorphic association.
So not sure, but I do not see better solution :disappointed:

Reply:
ok, so for the mapping we actually want to see only the explicit relations, not the calculated outcomes

given the definition complexity, I guess a test on such relation might be useful

Reply:
I have reversed it and putted a comment on the method, I dislike the behaviour about the two external groups - you can't know which group the user came from, so having a relation for uncertain thing, which should not be used anyway does not make sense xD